common/datasets/librispeech.py

```diff
@@ -0,0 +1,181 @@
+"""
+This file contains helper functions for the (common) pipeline steps needed to use the LibriSpeech corpus
+"""
+import os
+
+from recipe.i6_core.audio.encoding import BlissChangeEncodingJob
+from recipe.i6_core.corpus.transform import MergeCorporaJob, MergeStrategy
+from recipe.i6_core.datasets.librispeech import *
+from recipe.i6_core.meta.system import CorpusObject
+
+
+durations = {
+    "train-clean-100": 100.6,
+    "train-clean-360": 363.6,
+    "train-other-500": 496.7,
+    "dev-clean": 5.4,
+    "dev-other": 5.3,
+    "test-clean": 5.4,
+    "test-other": 5.1,
+}
+
+durations["train-clean-460"] = (
+    durations["train-clean-100"] + durations["train-clean-360"]
+)
+durations["train-other-960"] = (
+    durations["train-clean-460"] + durations["train-other-500"]
+)
+
+
+def get_bliss_corpus_dict(audio_format="flac", create_alias_with_prefix=None):
+    """
+    Download and create a bliss corpus for each of the LibriSpeech training corpora and test sets,
+    and return all corpora as a single corpus dict.
+
+    :param str audio_format: flac (no re-encoding), wav or ogg
+    :param str|None create_alias_with_prefix:
+    :return: A corpus dict with the following entries:
+        - 'dev-clean'
+        - 'dev-other'
+        - 'test-clean'
+        - 'test-other'
+        - 'train-clean-100'
+        - 'train-clean-360'
+        - 'train-clean-460'
+        - 'train-other-500'
+        - 'train-other-960'
+    :rtype: dict[str, Path]
+    """
+
+    create_alias_with_prefix = os.path.join(create_alias_with_prefix, "LibriSpeech")
+
+    download_metadata_job = DownloadLibriSpeechMetadataJob()
+    if create_alias_with_prefix:
+        download_metadata_job.add_alias(
+            os.path.join(create_alias_with_prefix, "download", "metadata")
+        )
+
+    def _get_corpus(corpus_name):
+        download_corpus_job = DownloadLibriSpeechCorpusJob(corpus_key=corpus_name)
+        create_bliss_corpus_job = LibriSpeechCreateBlissCorpusJob(
+            corpus_folder=download_corpus_job.out_corpus_folder,
+            speaker_metadata=download_metadata_job.out_speakers,
+        )
+        if create_alias_with_prefix:
```

Comment:
The variable name `create_alias_with_prefix` is misleading. It is a path, but the name reminds me of a boolean. And also Shouldn't we do a more specific check here? Just as an example, checking that it is not `None` and it has the substring `LibriSpeech` in it?

Reply:
I do not understand, "LibriSpeech" is always added, and it does check for `None`